docs/src/main/asciidoc/WildFly_Maven_Plugin_Guide.adoc

```diff
@@ -0,0 +1,410 @@
+
+Galleon is configured thanks to the Maven plugin ```<configuration>``` element.
+
+The first required piece of information that Galleon needs is a reference to the WildFly Galleon feature-pack. 
+The WildFly Galleon feature-pack is a maven artifact that contains everything needed to dynamically provision a server. 
+This feature-pack, as well as the feature-packs on which its depends,
+are deployed in public maven repositories. 
+
+At plugin execution time the WildFly feature-packs are retrieved and their content is assembled to provision a server.
+
+Once you have identified a WildFly Galleon feature-pack, you need to select a set of 
+link:#wildfly_layers[WildFly Layers] that are used to compose the server.
+
+The set of Galleon layers to include is driven by the needs of the application you are developing.
+The list of link:#wildfly_layers[WildFly Layers] provides details on 
+the server features that each layer brings. Make sure that the API and server features you are using (eg: Jakarta RESTful Web Services, MicroProfile Config, datasources)
+are provided by the layers you are choosing.
+
+If you decide not to specify Galleon layers, a complete server (WildFly server zip content) is provisioned.
+
+Maven Plugin configuration extract example:
+
+[source,xml]
+----
+<build>
+  <plugins>
+    <plugin>
+      <groupId>org.wildfly.plugins</groupId>
+      <artifactId>wildfly-maven-plugin</artifactId>
+      <configuration>
+        <feature-packs>
+          <feature-pack>
+            <location>wildfly@maven(org.jboss.universe:community-universe)</location> (1)
+          </feature-pack>
+        </feature-packs>
+        <layers>
+          <layer>jaxrs-server</layer> (2)
+        </layers>
+      </configuration>
+      <executions>
+        <execution>
+          <goals>
+            <goal>package</goal>
+          </goals>
+        </execution>
+      </executions>
+    </plugin>
+  </plugins>
+</build>
+----
+
+(1) In this plugin configuration extract, we are retrieving the latest WildFly Galleon feature-pack installed in the
+ ```org.jboss.universe:community-universe``` Galleon universe. In case you would like to provision a specific version of the server,
+you would need to specify the server version, for example ```wildfly@maven(org.jboss.universe:community-universe)#33.0.0.Final```
+
+(2) The link:#gal.jaxrs-server[jaxrs-server layer] and all its dependencies are provisioned.
+
+[NOTE]
+====
+
+If your project is using link:WildFly_and_WildFly_Preview{outfilesuffix}[WildFly Preview], the ``location`` to use
+is ``wildfly-preview@maven(org.jboss.universe:community-universe)``.
+
+====
+
+==== Specifying additional JBoss Modules modules
+
+In general, the JBoss Modules modules your application requires are provisioned by the specified Galleon layers. 
+In some very special cases, for example the REST XXX, you can explicitly add some JBoss Modules modules:
```

Comment:
Is the "XXX" here just a place holder?

Reply:
Yes, a place holder, as soon as we have the RESTeasy part, I will link it from there.

Reply:
For now I suggest:

'To handle some special cases -- for example some uses of Jakarta REST where optional RESTEasy modules are not provided by the `jax-rs' layer -- you can explicitly add some JBoss Modules modules:"

We can follow this up with a link later but I don't want to try and coordinate 2 changes at this stage of WF 34 work.

Reply:
Thank-you. I implemented your suggestion.